astropy/utils/misc.py

```diff
@@ -159,22 +163,28 @@ class NumpyRNGContext:
 
     """
 
-    def __init__(self, seed):
+    def __init__(self, seed: int) -> None:
         self.seed = seed
 
-    def __enter__(self):
-        from numpy import random
-
-        self.startstate = random.get_state()
-        random.seed(self.seed)
-
-    def __exit__(self, exc_type, exc_value, traceback):
-        from numpy import random
+    def __enter__(self) -> None:
+        self.startstate = np.random.get_state()
+        np.random.seed(self.seed)
 
-        random.set_state(self.startstate)
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: TracebackType | None,
+    ) -> None:
+        np.random.set_state(self.startstate)
 
 
-def find_api_page(obj, version=None, openinbrowser=True, timeout=None):
+def find_api_page(
+    obj: object,
+    version: str | None = None,
+    openinbrowser: bool = True,
+    timeout: float | None = None,
+) -> str:
```

Comment:
Flagging for follow-up for replacing the output with the more structured `urllib.parse.ParseResult`.

Reply:
But why?

Reply:
Or maybe a  https://docs.python.org/3/library/urllib.request.html#urllib.request.Request object is better. 
Not to sound tautological but IMO structured data should be structured. Especially in a OOPL. URLs aren't strings, though they can be represented by them. Like file paths aren't strings, but can be represented as such. Basically if something requires parsing and validation then it's better to do that once and return a structured output.

Reply:
There's a wonderful essay on "parse" not "validate" (having trouble finding it atm) that talks about this stuff.

Reply:
But it will break API and no one asked for it. TBH I don't even know if anyone really uses it. Anyways, I see you opened an issue, so I'll stop here.